addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/rest/RestClientImpl.java

```diff
@@ -0,0 +1,152 @@
+public class RestClientImpl implements RestClient {
+
+    private Logger logger = LoggerFactory.getLogger(RestClientImpl.class.getName());
+
+    private HttpClient httpClient;
+    private BundleContext bundleContext;
+    private String baseURL;
+    private int timeout;
+
+    public void activate(BundleContext bundleContext) throws NoSuchAlgorithmException {
+        this.bundleContext = bundleContext;
+
+        httpClient = new HttpClient(new SslContextFactory());
+
+        loadCertificate(CERTIFICATE_DIR, CERTIFICATE_FILE_NAME);
+        setBaseURL(MIHOME_REST_API_URL);
+        setConnectionTimeout(DEFAULT_REQUEST_TIMEOUT);
+        startHttpClient();
+    }
+
+    public void deactivate(BundleContext context) {
+        stopHttpClient();
+    }
+
+    @Override
+    public synchronized ContentResponse sendRequest(String requestPath, HttpMethod httpMethod, Properties httpHeaders,
+            InputStream content, String contentType) throws IOException {
+
+        String url = getBaseURL() + "/" + requestPath;
+        Request request = httpClient.newRequest(url).method(httpMethod).timeout(getConnectionTimeout(),
+                TimeUnit.MILLISECONDS);
+
+        if (httpHeaders != null) {
+            for (String httpHeaderKey : httpHeaders.stringPropertyNames()) {
+                request.header(httpHeaderKey, httpHeaders.getProperty(httpHeaderKey));
+            }
+        }
+
+        // add content if a valid method is given ...
+        if (content != null && (httpMethod.equals(HttpMethod.POST) || httpMethod.equals(HttpMethod.PUT))) {
+            request.content(new InputStreamContentProvider(content), contentType);
+        }
+
+        logger.debug("About to execute {}", request.getURI());
+        ContentResponse response = null;
+        try {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+
+            throw new IOException(e);
+        }
+        return response;
+    }
+
+    private void loadCertificate(String relativePath, String file) {
+        SSLContext sslContext = SSLContextBuilder.create(bundleContext).withTrustedCertificate(relativePath, file)
+                .build();
+        if (sslContext != null) {
+            httpClient.getSslContextFactory().setSslContext(sslContext);
+        } else {
+            logger.error("Not able to load Mi|Home certificate. The connection with the server may not be secure.");
+        }
+
+    }
+
+    @Override
+    public void setSslContext(SSLContext sslContext) {
+        // The SSLContext can be replaced when the client is stopped
+        stopHttpClient();
+        httpClient.getSslContextFactory().setSslContext(sslContext);
+        startHttpClient();
+
+    }
+
+    private void startHttpClient() {
+        if (!httpClient.isStarted()) {
+            try {
+                httpClient.start();
+            } catch (Exception e) {
+                logger.warn("Cannot start HttpClient!", e);
```

Comment:
imho you should not (ignore, because logging and returning normally is ignore the fact that went wrong) any exception in that case. Because if some exception occurs you are done there is nothing you can do to recover, otherwise they would have mentioned in their documentation. 

So the activate method should forward the exception because you failed to activate the RestClient.